### PR TITLE
bpo-34061: Add missing docs for NotSupportedError sqlite module

### PR DESCRIPTION
**Pull request by n0npax**
Add missing docs for NotSupportedError in sqlite module

https://bugs.python.org/issue34061

### CODE REVIEW

File: Doc/library/sqlite3.rst [hunk 1]
```diff
@@ -828,6 +828,12 @@ Exceptions
    disconnect occurs, the data source name is not found, a transaction could
    not be processed, etc.  It is a subclass of :exc:`DatabaseError`.
 
+.. exception:: NotSupportedError
+
+   Exception raised in case a method or database API was used which is not
+   supported by the database, e.g. requesting a .rollback() on a connection
+   that does not support transaction or has transactions turned off.
+   It must be a subclass of :exc:`DatabaseError`.
```
Comment (berkerpeksag) [inline Doc/library/sqlite3.rst]: It must be -> It is (PEP 249 is basically a specification, and it has already been implemented in the sqlite3 module, so we don't need use "must be".)

File: Doc/library/sqlite3.rst [hunk 2]
```diff
@@ -828,6 +828,12 @@ Exceptions
    disconnect occurs, the data source name is not found, a transaction could
    not be processed, etc.  It is a subclass of :exc:`DatabaseError`.
 
+.. exception:: NotSupportedError
+
+   Exception raised in case a method or database API was used which is not
+   supported by the database, e.g. requesting a .rollback() on a connection
```
Comment (berkerpeksag) [inline Doc/library/sqlite3.rst]: Perhaps we can replace "requesting a .rollback() on" with something like "calling the :meth:\`Connection.rollback\` method on" to make it sound more like stdlib documentation, not PEP 249?

File: Misc/ACKS [hunk 3]
```diff
@@ -1784,6 +1784,7 @@ Gordon Worley
 Darren Worrall
 Thomas Wouters
 Daniel Wozniak
+Marcin Niemira
```
Comment (berkerpeksag) [inline Misc/ACKS]: Thank you for adding your name to the list, but I usually prefer adding a contributor's name for more complicated patches (or after a couple of trivial patches)

**Review by berkerpeksag — CHANGES_REQUESTED** [review]
Thank you for the PR! :) I think this looks pretty good, I just left some comments mostly on documentation style.

File: Doc/library/sqlite3.rst (hunk 1 above)
Comment (n0npax) [reply Doc/library/sqlite3.rst]: Fixed

File: Doc/library/sqlite3.rst (hunk 2 above)
Comment (n0npax) [reply Doc/library/sqlite3.rst]: Fixed. Link works

File: Misc/ACKS (hunk 3 above)
Comment (n0npax) [reply Misc/ACKS]: Already merged by other PR.

**Comment by n0npax** [comment]
Hi, I've fixed requested changes. Thanks for Review.

File: Doc/library/sqlite3.rst [hunk 4]
```diff
@@ -835,6 +835,13 @@ Exceptions
    disconnect occurs, the data source name is not found, a transaction could
    not be processed, etc.  It is a subclass of :exc:`DatabaseError`.
 
+.. exception:: NotSupportedError
+
+   Exception raised in case a method or database API was used which is not
+   supported by the database, e.g. calling the :meth:`Connection.rollback`
```
Comment (berkerpeksag) [inline Doc/library/sqlite3.rst]: missing "method"

Comment (berkerpeksag) [reply Doc/library/sqlite3.rst]: I think :meth:\`~Connection.rollback\` would be a better choice here.

Comment (n0npax) [reply Doc/library/sqlite3.rst]: added missing "method" word & changed ```:meth:`Connection.rollback``` to the ```:meth:`~Connection.rollback```

**Review by berkerpeksag — APPROVED** [review]

**Comment by berkerpeksag** [comment]
Thanks!

**Comment by berkerpeksag** [comment]
@n0npax would you like to submit backport PRs?

**Comment by n0npax** [comment]
PR prepared.
@berkerpeksag thanks for corrections.

**Pull request merged**